### PR TITLE
Add known_data_files and a few more config files

### PR DESCRIPTION
**Pull request by usama8800**
Adds support for `programs.plasma.dataFile` in `rc2nix.py`

### CODE REVIEW

**Comment by magnouvean** [comment]
Just want to say I've tested and it seems to work fine on my end at the very least. The code looks good as well so I'd be happy to merge.

**Review by HeitorAugustoLN — APPROVED** [review]
Works flawlessy, just needs to be formatted, because of #370

File: script/rc2nix.py [hunk 1]
```diff
@@ -97,21 +115,27 @@ def __init__(self, file_name: str):
         def parse(self):
 
             def is_group_line(line: str) -> bool:
-                return re.match(r'^\s*(\[[^\]]+\]){1,}\s*$', line) is not None
+<<<<<<< HEAD
+                return re.match(r"^\s*(\[[^\]]+\])+\s*$", line) is not None
+=======
+                return re.match(r'^\s*(\[[^\]]+\])+\s*$', line) is not None
+>>>>>>> 670eb4a1fa1c4931685c854f6c47ed9a485152cc
```
Comment (HeitorAugustoLN) [inline script/rc2nix.py]: Merge conflict need to be resolved

File: script/rc2nix.py [hunk 2]
```diff
@@ -150,75 +180,115 @@ def should_skip_by_lambda(group: str, key: str) -> bool:
...
             print("    enable = true;")
             print("    shortcuts = {")
-            print(self.pp_shortcuts(settings.get("kglobalshortcutsrc", {}), 6))
+<<<<<<< HEAD
+            print(
+                self.pp_shortcuts(self.config_settings.get("kglobalshortcutsrc", {}), 6)
+            )
+=======
+            print(self.pp_shortcuts(self.config_settings.get("kglobalshortcutsrc", {}), 6))
+>>>>>>> 670eb4a1fa1c4931685c854f6c47ed9a485152cc
```
Comment (HeitorAugustoLN) [inline script/rc2nix.py]: Merge conflict need to be resolved here also

**Comment by HeitorAugustoLN** [comment]
Thanks for your contribution! 😃

**Pull request merged**